tests/integration/test_remote_write_with_zinc.py

```diff
@@ -13,10 +13,56 @@
 
 prometheus_name = "prometheus"
 agent_name = "grafana-agent"
+zinc_name = "zinc"
+apps = [prometheus_name, agent_name, zinc_name]
 
 
 @pytest.mark.abort_on_fail
-async def test_create_remote_write_models(ops_test, prometheus_charm):
+async def test_remote_write_with_zinc(ops_test, prometheus_charm, zinc_charm):
+    """Test that Prometheus can be related with the Grafana Agent over remote_write."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            prometheus_charm,
+            resources={"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")},
+            application_name=prometheus_name,
+            trust=True,  # otherwise errors on ghwf (persistentvolumeclaims ... is forbidden)
+        ),
+        ops_test.model.deploy(
+            "grafana-agent-k8s",
+            application_name=agent_name,
+            channel="edge",
+        ),
+        ops_test.model.deploy(
+            zinc_charm,
+            resources={"zinc-image": "jnsgruk/zinc:0.1.9"},
+            application_name=zinc_name,
+        ),
+    )
+
+    await ops_test.model.wait_for_idle(apps=apps, status="active", wait_for_units=1)
+    assert await check_prometheus_is_ready(ops_test, prometheus_name, 0)
+
+    await asyncio.gather(
+        ops_test.model.add_relation(prometheus_name, agent_name),
+        ops_test.model.add_relation(
+            f"{agent_name}:metrics-endpoint", f"{zinc_name}:metrics-endpoint"
+        ),
+    )
+
+    await ops_test.model.wait_for_idle(apps=apps, status="active", idle_period=90)
+
+    assert await has_metric(
+        ops_test,
+        f'up{{juju_model="{ops_test.model_name}",juju_application="{agent_name}"}}',
+        prometheus_name,
+    )
+
+    await ops_test.model.reset()
+    await ops_test.model.wait_for_idle(idle_period=90)
```

Comment:
- Is reset much faster than a separate test file?
- Can we consolidate this function and the other `test_remote_write_*`?